src/transmission_manager.rs

```diff
@@ -82,48 +90,45 @@ fn infection_attempt(
 }
 
 fn get_next_infection_time_unif(context: &mut Context, last_infection_time_unif: f64) -> f64 {
-    // this is NOT order statistics
-    // we are just playing around to make a point
-    // we will update the math to be correct later
+    // Get the next infection time by sampling from a uniform distribution.
+    // FOR NOW, we are using placeholder math to guarantee the infection times are always increasing.
+    // This is not order statistics. This will be corrected at a later time.
     context.sample_range(TransmissionRng, last_infection_time_unif..1.0)
 }
 
-fn get_next_infection_time_from_gi(
-    _context: &mut Context,
-    gi: f64,
-    next_infection_time_unif: f64,
-) -> f64 {
-    // this will be properly stored later
-    //let gi = context.get_data_container(generation_interval).unwrap();
-    // the math here is wrong as well -- we are not implementing order statistics
-    // there is no guarantee that these infection attempts are sequential draws from the GI distribution
-    // this will be fixed with real math later
+fn get_next_infection_time_from_gi(context: &mut Context, next_infection_time_unif: f64) -> f64 {
+    // Calculate the next infection time by passing the next uniform random number
+    // through the inverse CDF of the generation interval (i.e., inverse transform
+    // sampling).
+    // In the future, we will generalize the use of an exponential distribution to
+    // an arbitrary distribution with a defined inverse CDF.
+    let gi = context
+        .get_global_property_value(Parameters)
+        .unwrap()
+        .generation_interval;
     Exp::new(1.0 / gi)
         .unwrap()
         .inverse_cdf(next_infection_time_unif)
 }
 
 fn schedule_next_infection_attempt(
     context: &mut Context,
-    transmitee_id: PersonId,
-    gi: f64,
-    num_infection_attempts_remaining: f64,
+    transmitter_id: PersonId,
+    num_infection_attempts_remaining: usize,
     last_infection_time_unif: f64,
 ) {
-    // get next infection attempt time
+    // Get the next infection attempt time, which is greater than the last infection attempt time.
```

Comment:
I would just say "Get the next infection time". The greater than part is enforced elsewhere.